capnproto: use upstream patch to fix w/musl

dtzWill opened the pull request:
###### Description of changes

Fix capnproto on musl.  Note that the package builds either way, but as-is will fail when linking (perhaps when using certain functionality) due to generated references to `getcontext` and friends.

Fix the situation by using an upstream patch, vendor'd to adjust so that it applies.

Resolves issue in downstream project.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

dtzWill committed:
capnproto: use upstream patch to fix w/musl

Without this, the package builds but fails when linking
sometimes due to undefined references to `getcontext` and friends.

Fix this using an upstream patch providing musl support,
which is vendored so it can be adjusted to apply to the release.

risicle reviewed (approved):
WFM, nixos x86_64.

Personally I tend to put patch comments in the patch header as technically that patch is no longer 85b4e6074882d62593ae90d6683e24efccb045f0 and I fear an original author being upset that I'm presenting a botched version of the patch as theirs..

dtzWill committed:
capnproto: regenerate patch with more clear commit message

Explain what changed, and ensure no one mistakes this for the original
upsteam commit (while giving due attribution).

dtzWill commented:
> Personally I tend to put patch comments in the patch header as technically that patch is no longer 85b4e6074882d62593ae90d6683e24efccb045f0 and I fear an original author being upset that I'm presenting a botched version of the patch as theirs..

I agree, and thanks for pushing on this.  If nothing else it's confusing to find a patch claiming to be a certain commit and find out it's actually not, and /that's/ why it wasn't working as expected or whatever.  Misleading if not a little dishonest, especially with upstream author names on it (but also makes sense for attribution reasons...).  Commit signing kinda addresses this? Anyway FWIW actually 85b4e6074882d62593ae90d6683e24efccb045f0 is my commit (to a local unpublished fork for doing the cherry-pick to v0.9.1), so no worries directly about being confused as an upstream 85b4e6.  The original is the mentioned cherry-picked revision 77ac9154440bcc216fda1092fd5bb51da62ae09c .

Anyway, updated patch to reflect all this more clearly, hopefully (and now it has a new commit hash).

Look good? :smile:

risicle commented:
LGTM.
